### PR TITLE
Fix a bug in the computation of geometric acceleration

### PR DESCRIPTION
**Pull request by pleroy**

### CODE REVIEW

File: physics/barycentric_rotating_dynamic_frame_test.cpp [hunk 1]
```diff
@@ -309,19 +313,21 @@ TEST_F(BarycentricRotatingDynamicFrameTest, CentrifugalAcceleration) {
   {
     InSequence s;
     EXPECT_CALL(*mock_ephemeris_,
-                ComputeGravitationalAcceleration(big_dof.position(), t))
+                ComputeGravitationalAccelerationOnMassiveBody(
+                    base::check_not_null(big_), t))
```
Comment (eggrobin) [inline physics/barycentric_rotating_dynamic_frame_test.cpp]: using

File: physics/barycentric_rotating_dynamic_frame_test.cpp [hunk 2]
```diff
@@ -464,5 +474,22 @@ TEST_F(BarycentricRotatingDynamicFrameTest, LinearAcceleration) {
...
+       Velocity<BigSmallFrame>({3 * Metre / Second,
+                                2 * Metre / Second,
+                                1 * Metre / Second})};
+  // We trust the functions to compute the values correctly, but this test
+  // ensures that we don't get NaNs.
+  EXPECT_THAT(big_small_frame_->GeometricAcceleration(t, point_dof),
+              AlmostEquals(Vector<Acceleration, BigSmallFrame>({
+                  2.32786248002527236E3 * Metre / Pow<2>(Second),
+                  -3.61670567977415587E1 * Metre / Pow<2>(Second),
+                  -5.38007972376415182e+01 * Metre / Pow<2>(Second)}), 0));
```
Comment (eggrobin) [inline physics/barycentric_rotating_dynamic_frame_test.cpp]: Consistent exponent case.

File: physics/mock_ephemeris.hpp [hunk 3]
```diff
@@ -48,13 +48,24 @@ class MockEphemeris : public Ephemeris<Frame> {
...
-  // compiler.
+  // NOTE(phl): This overload introduces ambiguities in the expectations.
+  // MOCK_CONST_METHOD2_T(
+  //     ComputeGravitationalAccelerationOnMasslessBody,
+  //     Vector<Acceleration, Frame>(
+  //         not_null<DiscreteTrajectory<Frame>*> /*const*/ trajectory,
+  //         Instant const& t));
+
+  // NOTE(phl): The commented-out const below is to work-around a compiler
+  // internal error.  Don't ask.
```
Comment (eggrobin) [inline physics/mock_ephemeris.hpp]: ...

File: physics/mock_ephemeris.hpp [hunk 4]
```diff
@@ -48,13 +48,24 @@ class MockEphemeris : public Ephemeris<Frame> {
...
+  MOCK_CONST_METHOD2_T(
+      ComputeGravitationalAccelerationOnMassiveBody,
+      Vector<Acceleration, Frame>(
+        not_null<MassiveBody const*> /*const*/ body,
```
Comment (eggrobin) [inline physics/mock_ephemeris.hpp]: badindent

File: physics/barycentric_rotating_dynamic_frame_test.cpp (hunk 1 above)
Comment (pleroy) [inline physics/barycentric_rotating_dynamic_frame_test.cpp]: Done.

File: physics/barycentric_rotating_dynamic_frame_test.cpp (hunk 2 above)
Comment (pleroy) [inline physics/barycentric_rotating_dynamic_frame_test.cpp]: Done.

File: physics/mock_ephemeris.hpp (hunk 4 above)
Comment (pleroy) [inline physics/mock_ephemeris.hpp]: Done.

**Pull request merged**